deps/jerry/jerry-core/ecma/builtin-objects/ecma-builtins.c

```diff
@@ -769,6 +783,18 @@ ecma_builtin_try_to_instantiate_property (ecma_object_t *object_p, /**< object *
       func_obj_p = ecma_builtin_make_function_object_for_routine (builtin_id,
                                                                   ECMA_GET_ROUTINE_ID (curr_property_p->value),
                                                                   ECMA_GET_ROUTINE_LENGTH (curr_property_p->value));
+      lit_magic_string_id_t id = ecma_builtin_routine_get_name (builtin_id, ECMA_GET_ROUTINE_ID (curr_property_p->value));
+      if (id != magic_string_id)
+      {
+        /* These builtins use custom routine dispatcher instead of template.
+         *  So routine id are not consist to declaration in their inc.h.
+         *  Magic string id of name should be set correctly.
+         */
+        JERRY_ASSERT (builtin_id ==ECMA_BUILTIN_ID_DATE_PROTOTYPE || builtin_id == ECMA_BUILTIN_ID_MATH );
+        ecma_builtin_routine_set_name (builtin_id,
+                                                                    ECMA_GET_ROUTINE_ID (curr_property_p->value),
+                                                                    magic_string_id);
+      }
       value = ecma_make_object_value (func_obj_p);
```

Comment:
If re-get id by ecma_builtin_routine_get_name and assert id == magic_string_id, test262 will pass all.
So I think test coverage is enough.